tests/Unit/CartTest.php

```diff
@@ -0,0 +1,73 @@
+<?php
+use PHPUnit\Framework\TestCase;
+use App\Cart;
+
+class CartTest extends TestCase
+{
+    /**
+     * If item is added to the cart
+     *
+     * @covers Cart::addItem
+     */
+    public function testAddItem()
+    {
+        $cart = new Cart();
+        $cart->addItem('foo');
+        $this->assertEquals(['foo'], $cart->getItems());
+        $this->assertNotEquals(['bar'], $cart->getItems());
+    }
+
+    /**
+     * Remove an item from the cart
+     *
+     * @covers Cart::removeItem
+     */
+    public function testRemoveItem()
+    {
+        $cart = new Cart(['foo']);
+        $cart->removeItem('foo');
+        $this->assertEquals([], $cart->getItems());
+
+        $cart = new Cart(['foo', 'bar', 'horse', 3]);
+        $cart->removeItem(3);
+        $this->assertEquals(['foo', 'bar', 'horse'], $cart->getItems());
+
+        $cart = new Cart(['foo', 'bar', 'horse', 'bar', 'horse', 3]);
+        $cart->removeItem('foo');
+        $this->assertEquals(['bar', 'horse', 'bar', 'horse', 3], $cart->getItems());
+    }
+
+    /**
+     * Get the list of items in the cart
+     *
+     * @covers Cart::getItems
+     */
+    public function testGetItems()
+    {
+        $cart = new Cart();
+        $this->assertEquals([], $cart->getItems());
+
+        $cart = new Cart([1, 2, 3]);
+        $this->assertEquals([1, 2, 3], $cart->getItems());
+    }
+
+    /**
+     * Get the list of items, as a number, in the cart
+     *
+     * @covers Cart::getItems
+     * @depends testGetItems
+     */
+    public function testGetItemsCount()
+    {
+        $cart = new Cart([0, 1, 2, 3]);
+        $this->assertEquals(4, $cart->getItemsCount());
+    }
+
+    public function testUpdateQuantity()
+    {
+        $cart = new Cart(['foo']);
+        $this->assertEquals(['foo'], $cart->getItems());
+        $cart->updateQuantity('foo', 6);
+        $this->assertEquals(['foo', 'foo', 'foo', 'foo', 'foo', 'foo', 'foo'], $cart->getItems());
```

Comment:
@AC-TimRourke 
In this example here where I'm just focusing on the logic for the `updateQuantity` method, I am able to get the result from the logic that I had in mind, but I am unable to update my array to filter out the rest of the `foo` item and only keep 5 `foo`. I have a total of `7` again.

#### Code
```php
$items = ['foo', 'bar', 'foo'];
$item = ['foo'];
$qty = 5;

$filteredItems = array_filter( $items, function($value) use ($item) { return $value === $item; } );
$updatedQuantity = array_diff($items, $filteredItems);

for ($i = 0; $i < $qty; $i++) {
  $updatedQuantity[] = $item;
}
  
print_r($updatedQuantity);
```

#### Output
```html
Array
(
    [0] => foo
    [1] => bar
    [2] => foo
    [3] => Array
        (
            [0] => foo
        )

    [4] => Array
        (
            [0] => foo
        )

    [5] => Array
        (
            [0] => foo
        )

    [6] => Array
        (
            [0] => foo
        )

    [7] => Array
        (
            [0] => foo
        )

)
```

Here is my code snippet to play with: https://repl.it/Jr3n